luxonis_train/loaders/luxonis_perlin_loader_torch.py

```diff
@@ -0,0 +1,92 @@
+import glob
+import os
+import random
+
+import numpy as np
+import torch
+from torch import Tensor
+
+from luxonis_train.enums import TaskType
+
+from .luxonis_loader_torch import LuxonisLoaderTorch
+from .perlin import apply_anomaly_to_img
+
+
+class LuxonisLoaderPerlinNoise(LuxonisLoaderTorch):
+    def __init__(
+        self,
+        *args,
+        anomaly_source_path: str | None = None,
+        noise_prob: int = 0.5,
+        mean: list | None = None,
+        std: list | None = None,
+        **kwargs,
+    ):
+        """Custom loader for Luxonis datasets that adds Perlin noise
+        during training with a given probability.
+
+        @type anomaly_source_path: str | None
+        @param anomaly_source_path: Path to the anomaly dataset from
+            where random samples are drawn for noise.
+        @type noise_prob: int
+        @param noise_prob: The probability with which to apply Perlin
+            noise (only used during training).
+        @type mean: list
+        @param mean: The mean values for the image normalization.
+        @type std: list
+        @param std: The standard deviation values for the image
+            normalization.
+        """
+        super().__init__(*args, **kwargs)
+        if anomaly_source_path and os.path.exists(anomaly_source_path):
+            self.anomaly_source_paths = sorted(
+                glob.glob(os.path.join(anomaly_source_path, "*/*.jpg"))
+            )
+            if not self.anomaly_source_paths:
+                raise FileNotFoundError(
+                    "No .jpg files found at the specified path."
+                )
+        else:
+            raise ValueError("Invalid or unspecified anomaly source path.")
+        self.anomaly_source_path = anomaly_source_path
+        self.noise_prob = noise_prob
+        if mean is None:
```

Comment:
we should have access to the Augmentation instance here. Can we get `Normalize` augmentation from there and apply it on the noise image as well instead of requiring mean and std params? And if Normalize isn't present then don't normalize.